From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Markus Mayer <mmayer@broadcom.com>
Date: Fri, 29 May 2026 10:15:13 -0700
Subject: [PATCH] t_chmod_secure: use HAVE_OPENAT2 to check for openat2()
 support

To prevent using openat2() in situations where it is not supported, use
    #if defined(__linux__) && defined(HAVE_OPENAT2)
in t_chmod_secure.c, just like it was already being done in syscall.c.

Signed-off-by: Markus Mayer <mmayer@broadcom.com>
---
 t_chmod_secure.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/t_chmod_secure.c b/t_chmod_secure.c
index 7c57dbbc..130c7419 100644
--- a/t_chmod_secure.c
+++ b/t_chmod_secure.c
@@ -17,7 +17,7 @@
 
 #include <sys/stat.h>
 
-#ifdef __linux__
+#if defined(__linux__) && defined(HAVE_OPENAT2)
 #include <sys/syscall.h>
 #include <linux/openat2.h>
 #endif
@@ -45,7 +45,7 @@ static int errs = 0;
 static int kernel_resolve_beneath_supported(void)
 {
 	int fd;
-#ifdef __linux__
+#if defined(__linux__) && defined(HAVE_OPENAT2)
 	{
 		struct open_how how;
 		memset(&how, 0, sizeof how);
